distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/NsiliEndpointTest.java

```diff
@@ -0,0 +1,241 @@
+            fail("Failed in @BeforeExam: " + e.getMessage());
+        }
+    }
+
+    @AfterExam
+    public void stopHttpListener() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Before
+    public void startSampleNsiliClientFeature() throws Exception {
+        getServiceManager().startFeature(true, "sample-nsili-client");
+    }
+
+    @After
+    public void stopSampleNsiliClientFeature() throws Exception {
+        getServiceManager().stopFeature(true, "sample-nsili-client");
+    }
+
+    @Test
+    public void testStandingQueryMgr() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        sampleNsiliClient.testStandingQueryMgr();
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrGetHitCount() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        assertThat(sampleNsiliClient.getHitCount(), is(1));
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testSubmitQuery() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        // CatalogMgr
+        DAG[] results = sampleNsiliClient.submit_query();
+        assertThat(results.length, is(1));
+        DAG result = results[0];
+        assertThat(sampleNsiliClient.getProductIdFromDag(result), is(ingestedProductId));
+
+        // ProductMgr
+        Map<String, String> parameters = sampleNsiliClient.get_parameters(result);
+        assertThat(parameters, IsMapContaining.hasEntry("identifier", ingestedProductId));
+        assertThat(parameters, IsMapContaining.hasEntry("status", "NEW"));
+        assertThat(sampleNsiliClient.get_related_file_types(result), is(equalTo(new String[]{"THUMBNAIL"})));
+        final String expectedThumbnailFilename = ingestedProductId + "-THUMBNAIL.jpg";
+        assertThat(sampleNsiliClient.get_related_files(result), is(equalTo(new String[]{expectedThumbnailFilename})));
+        assertStubServerPutRequest(expectedThumbnailFilename, "image/jpeg");
+
+        // OrderMgr
+        String orderResultFilename = sampleNsiliClient.testOrder(result);
+        // TODO test packageElements of the order response https://jira.di2e.net/browse/GDIB-1751
+        assertStubServerPutRequest(orderResultFilename, "application/x-tar");
+
+        deleteMetacard(ingestedProductId);
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrViaCallback() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
```

Comment:
Does the sampleNsiliClient need to be started new for each test, or can it be moved into an @Before?